src/bindings/python/src/compatibility/openvino/__init__.py

```diff
@@ -2,4 +2,68 @@
 # Copyright (C) 2018-2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-__path__ = __import__("pkgutil").extend_path(__path__, __name__)  # type: ignore # mypy issue #1422
+__path__ = __import__("pkgutil").extend_path(__path__, __name__)
+
+try:
+    # Internal modules
+    from openvino import test_utils as test_utils
+except ImportError:
+    pass
+
+try:
+    # Import all public modules
+    from openvino import runtime as runtime
+    from openvino import frontend as frontend
+    from openvino import helpers as helpers
+    from openvino import preprocess as preprocess
+    from openvino import utils as utils
+    from openvino.runtime import properties as properties
+    # Import old API
+    # TODO: remove in 2024.0
+    from openvino import inference_engine as inference_engine
+
+    # Required for Windows OS platforms
+    from openvino.utils import _add_openvino_libs_to_search_path
+
+    _add_openvino_libs_to_search_path()
+
+    # Import most important classes and functions from openvino.runtime
+    from openvino.runtime import Model
+    from openvino.runtime import Core
+    from openvino.runtime import CompiledModel
+    from openvino.runtime import InferRequest
+    from openvino.runtime import AsyncInferQueue
+
+    from openvino.runtime import Extension
+    from openvino.runtime import Dimension
+    from openvino.runtime import Strides
+    from openvino.runtime import PartialShape
+    from openvino.runtime import Shape
+    from openvino.runtime import Layout
+    from openvino.runtime import Type
+    from openvino.runtime import Tensor
+
+    from openvino.runtime import compile_model
+    from openvino.runtime import get_batch
+    from openvino.runtime import set_batch
+    from openvino.runtime import serialize
+    from openvino.runtime import shutdown
+    from openvino.runtime import tensor_from_file
+    from openvino.runtime import save_model
+
+    # Set version for openvino package
+    from openvino.runtime import get_version
+    __version__ = get_version()
+except ImportError:
+    import warnings
+    warnings.warn("openvino package has problems with imports!", ImportWarning, stacklevel=2)
+
+# Import openvino.tools
+# Capture it in try-except with pass so circular imports are allowed.
+# TODO: restructure packages to remove WA
```

Comment:
Is it correct understanding that try-except here is needed only to solve circular imports? Why do we have that circular imports and how to avoid it? @popovaan, could you also comment?

Reply:
looks like OVC internally import OV runtime and I expect infinite recursive here rather than circular dependency

Reply:
It was added in earlier stages and it fixed the problem. In current solution (as Ilya says) it creates infinite recursion as you are able to go `openvino.tools.openvino.tools.openvino...`.

However try-except here will not change much as it will stay here or to move one "try-except" above. OVC is part of `openvino` itself so it will be either imported correctly or passed around. Example would be using legacy MO from PYTHONPATH. So all inits must match and go around OVC as it won't be discovered and result in error in such case.